sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -321,7 +346,7 @@ public virtual Response<ConfigurationSetting> SetConfigurationSetting(Configurat
             }
         }
 
-        private Request CreateSetRequest(ConfigurationSetting setting, bool onlyIfChanged)
+        private Request CreateSetRequest(ConfigurationSetting setting, MatchConditions conditions)
```

Comment:
This introduces the question -- if ETag is populated in both `setting` and `conditions`, which takes precedence?  We avoided having to address this by using the convenience overload.

Reply:
The logic is the same as in `Set` without the MatchConditions, we don't do anything with the ETag. It's a purely readonly value provided by the service.

Reply:
IIRC, the most common use case here was that a customer would retrieve a setting, hold it in a cache, and then make updates to it and send it back to the service.  If they did that and specified a different ETag in MatchConditions, I think it would be confusing regarding what the intended behavior should be.

Reply:
I'm not sure that I fully understand. The customer clearly says "use these values for match conditions", why would they be confused about which value is being used?